static/js/show-task.js

```diff
@@ -39,20 +41,56 @@ function saveTask() {
     }});
 }
 
+function rememberState() {
+	// Finding how many number of rows are being displayed so that we can make it same after refresh
+        var dropdownElements = $('.page-list .dropdown-menu .dropdown-item');
+        var activeDropdownElement = $('.page-list .dropdown-menu .dropdown-item.active')[0];
+        for(var i=0; i < dropdownElements.length; i++)
+        {
+            if (dropdownElements[i] === activeDropdownElement) {
+                activeDropdownElementIndex = i;
+            }
+        }
+
+        // Finding which page is active so that we can make it active again after refresh
+        var paginationElements = $('.pagination .page-item .page-link');
+        var activeElement = $('.pagination .page-item.active .page-link')[0];
+        for(var i=0; i < paginationElements.length; i++)
+        {
+            if (paginationElements[i] === activeElement) {
+                activeElementIndex = i;
+            }
+        }
+}
+
+function restoreState() {
+      // making the number of rows to be same
+      var dropdownElements = $('.page-list .dropdown-menu .dropdown-item');
+      dropdownElements[activeDropdownElementIndex].click();
+
+      // making the same page active as was before
+      var paginationElements = $('.pagination .page-item .page-link');
+      paginationElements[activeElementIndex].click();
+}
+
+
 function refresh() {
     search_term = $("#search_by_task_name_form input[name=search_term]").val();
     worklist_name = $("#search_by_task_name_form input[name=worklist_name]").val();
     worklist_created_by = $("#search_by_task_name_form input[name=worklist_created_by]").val();
 
+    rememberState();
+
     $('#task_table').html('');
     $('#task_table').addClass('loader');
     $('#task_table').html('').load(
-    "/worklist-tool/update-task-table?search_term=" + encodeURIComponent(search_term) + "&worklist_name=" +
-        encodeURIComponent(worklist_name) + "&worklist_created_by=" + encodeURIComponent(worklist_created_by),
-    function() {
-      $('#task_table').removeClass('loader');
-    }
+    "/worklist-tool/update-task-table/" + encodeURIComponent(worklist_created_by) + "/" + encodeURIComponent(worklist_name) + "?search_term=" + encodeURIComponent(search_term),
```

Comment:
I can file a separate issue later, but just a FYI -- when testing this, I've seen this request take as long as a minute for my Surlycyborg/Colombia worklist, and during that time all the UI shows is the progress indicator.

We should figure out how to make this faster for large worklists, and (probably more important) make the request in the background and only update the table after it has completed instead of having the user wait for it. Unfortunately database reads on Toolforge can take a while sometimes...

Reply:
I've made a few changes to solve this problem. I've now written it in such a way that user will not have to wait for as long as the ajax call would take to fetch refreshed table, but will only have to wait for 1 sec (can be changed to any value), and in that 1 sec, a loader will appear on the screen as it does now, on an overlay with opacity 0.7, and at the same time table will be updated to have new values.